tests/test_workers/test_pkg_managers/test_golang.py

```diff
@@ -132,16 +128,21 @@ def test_resolve_gomod_no_deps(
         mock.Mock(returncode=0, stdout='github.com/release-engineering/retrodep/v2'),
     ]
     mock_golang_version.return_value = 'v2.1.1'
-    mock_path_exists.return_value = False
 
     archive_path = '/this/is/path/to/archive.tar.gz'
     request = {'id': 3, 'ref': 'c50b93a32df1c9d700e3e80996845bc2e13be848'}
     module, resolved_deps = resolve_gomod(archive_path, request)
 
     assert module == sample_package
     assert not resolved_deps
-    mock_makedirs.assert_called_once()
-    mock_add_deps.assert_called_once()
+    # The second one ensures the source cache directory exists
+    assert 2 == mock_path_mkdir.call_count
+
+    bundle_dir = RequestBundleDir(request['id'])
+    mock_copytree.assert_called_once_with(
+        str(tmpdir / RequestBundleDir.go_mod_cache_download_part),
```

Comment:
Optional: I prefer the `joinpath` syntax since it makes it clearer what is happening.